src/blocks/ErrorBoundary.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { ErrorInfo, ReactElement } from 'react'
+import { sendMessage } from '../communication/sendMessage'
+import { OutgoingExperienceBuilderEvent } from '../types'
+import './ErrorBoundary.css'
+
+class ImportedComponentError extends Error {}
+
+export class ErrorBoundary extends React.Component<
+  { children: ReactElement },
+  { hasError: boolean; error: Error | null; errorInfo: ErrorInfo | null; showErrorDetails: boolean }
+> {
+  constructor(props: { children: ReactElement }) {
+    super(props)
+    this.state = { hasError: false, error: null, errorInfo: null, showErrorDetails: false }
+  }
+
+  static getDerivedStateFromError() {
+    return { hasError: true }
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    this.setState({ error, errorInfo })
+    if (!(error instanceof ImportedComponentError)) {
+      sendMessage(OutgoingExperienceBuilderEvent.CANVAS_ERROR, error)
+    }
```

Comment:
Shall we still throw the error when it is an imported component error? So that we don't swallow it for the user

Reply:
So I updated the code to re-throw here. One thing though is that the error screen is no longer shown for any ImportedComponentError. If the customer have setup up their error error boundary in their website it will be handled there or a blank screen will be shown. But I guess this is ok because the original intention was to handle errors coming from our own code? What do you think?

Reply:
Yes I agree 👍